### PR TITLE
Introduce React component.

### PR DESCRIPTION
**Pull request by alexmacarthur**

### CODE REVIEW

File: packages/steppp-react/src/index.test.tsx [hunk 1]
```diff
@@ -0,0 +1,185 @@
+import { render, screen } from "@testing-library/react";
+import { describe, it, expect } from "vitest";
+import { Provider, Step, useSteppp, Wrapper } from ".";
+
+describe("Using standard implementation.", () => {
```
Comment (JZDesign) [inline packages/steppp-react/src/index.test.tsx:5]: # 🎸 

Good work on the tests. I'd like to see boundary testing. For example:

- Moving backwards from the 1st step does nothing
- Moving forwards from the last step does nothing

**Review by JZDesign — APPROVED** [review]

File: packages/steppp-react/README.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+Run `yarn add @ramseyinhouse/steppp-react` or `npm install @ramseyinhouse/steppp-react`.
+
+## Usage
+
+Import the `Provider`, `Wrapper`, and `Step` components from `@ramseyinhouse/steppp-react`. The `Provider` requires a child as a function that will provide methods for navigating through the experience.
+
+```tsx
+import { Provider, Wrapper, Step } from "@ramseyinhouse/steppp-react";
+
+const App = () => {
```
Comment (CalamityAdam) [inline packages/steppp-react/README.md:16]: 🎸 I know it's just an example, but it's a chance to promote better practices! 🚀

```suggestion
function App() {
```

File: packages/steppp-react/README.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+            <Step>three</Step>
+          </Wrapper>
+
+          <button onClick={forward}>Forward</button>
+          <button onClick={backward}>Backward</button>
+        </>
+      )}
+    </Provider>
+  );
+};
```
Comment (CalamityAdam) [inline packages/steppp-react/README.md:33]: 🎸 following above PR

```suggestion
}
```

File: packages/steppp-react/README.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+```
+
+### Context
+
+A `useSteppp()` hook exists for leveraging navigation methods within any component nested within the `Provider` component.
+
+```tsx
+import { useSteppp, Step } from "@ramseyinhouse/steppp-react";
+
+const SubStep = ({ children, ...rest }: { children: any }) => {
```
Comment (CalamityAdam) [inline packages/steppp-react/README.md:43]:
```suggestion
function SubStep({ children, ...rest }: { children: any }) {
```

File: packages/steppp-react/README.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  const { forward, backward, moveTo } = useSteppp();
+
+  return (
+    <Step {...rest}>
+      {children}
+
+      <button onClick={forward}>Move Forward</button>
+    </Step>
+  );
+};
```
Comment (CalamityAdam) [inline packages/steppp-react/README.md:53]:
```suggestion
}
```

File: packages/steppp-react/example/app.tsx [hunk 6]
```diff
@@ -0,0 +1,28 @@
+import React from "react";
+import { Provider, Wrapper, Step, useSteppp } from "../src/index";
+import SubStep from "./SubStep";
+
+const App = () => {
```
Comment (CalamityAdam) [inline packages/steppp-react/example/app.tsx:5]:
```suggestion
function App() {
```

File: packages/steppp-react/example/app.tsx [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+            <Step name="fourth">fourth</Step>
+          </Wrapper>
+
+          <button onClick={forward}>Forward</button>
+          <button onClick={backward}>Backward</button>
+        </>
+      )}
+    </Provider>
+  );
+};
```
Comment (CalamityAdam) [inline packages/steppp-react/example/app.tsx:26]:
```suggestion
}
```

File: packages/steppp-react/example/SubStep.tsx [hunk 8]
```diff
@@ -0,0 +1,13 @@
+import React from "react";
+import { Step, useSteppp } from "../src";
+
+export default ({ children }) => {
```
Comment (CalamityAdam) [inline packages/steppp-react/example/SubStep.tsx:4]: another great example of a place to promote good practices 💯 

```suggestion
function SubStep({ children }) {
```

File: packages/steppp-react/example/SubStep.tsx [hunk 9]
```diff
@@ -0,0 +1,13 @@
...
+  const { forward, backward, moveTo } = useSteppp();
+
+  return (
+    <Step>
+      {children}
+      <button onClick={() => forward()}>forward</button>
+    </Step>
+  );
+};
```
Comment (CalamityAdam) [inline packages/steppp-react/example/SubStep.tsx:13]:
```suggestion
}

export { SubStep };
```

File: packages/steppp-react/example/app.tsx [hunk 10]
```diff
@@ -0,0 +1,28 @@
+import React from "react";
+import { Provider, Wrapper, Step, useSteppp } from "../src/index";
+import SubStep from "./SubStep";
```
Comment (CalamityAdam) [inline packages/steppp-react/example/app.tsx:3]:
```suggestion
import { SubStep } from "./SubStep";
```

File: packages/steppp-react/example/app.tsx [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+
+          <button onClick={forward}>Forward</button>
+          <button onClick={backward}>Backward</button>
+        </>
+      )}
+    </Provider>
+  );
+};
+
+export default App;
```
Comment (CalamityAdam) [inline packages/steppp-react/example/app.tsx:28]:
```suggestion
export { App };
```

File: packages/steppp-react/example/index.tsx [hunk 12]
```diff
@@ -0,0 +1,11 @@
+import React from "react";
+import { createRoot } from "react-dom/client";
+import App from "./app";
```
Comment (CalamityAdam) [inline packages/steppp-react/example/index.tsx:3]:
```suggestion
import { App } from "./app";
```

File: packages/steppp-react/package.json [hunk 13]
```diff
@@ -0,0 +1,37 @@
+{
+    "name": "@ramseyinhouse/steppp-react",
+    "version": "0.0.1",
```
Comment (CalamityAdam) [inline packages/steppp-react/package.json:3]: arguably a deployed major version, no?

```suggestion
    "version": "1.0.0",
```

File: packages/steppp-react/src/components/Provider.tsx [hunk 14]
```diff
@@ -0,0 +1,54 @@
...
+  [key: string]: any;
+}
+
+export interface StepppInstance {
+  moveTo: (stepName: string) => void;
+  forward: () => void;
+  backward: () => void;
+}
+
+const baseCss = `
```
Comment (CalamityAdam) [inline packages/steppp-react/src/components/Provider.tsx:16]: oooh yeeeeaaaahhh 💯

File: packages/steppp-react/src/components/Provider.tsx [hunk 15]
```diff
@@ -0,0 +1,54 @@
...
+    }
+`;
+
+export function Provider({ children, ...rest }: WrapperProps) {
+  const stepRef = useRef<HTMLDivElement>(null);
+  const [stepppInstance, setStepppInstance] = useState<StepppInstance>(
+    null as unknown as StepppInstance
+  );
+
+  useLayoutEffect(() => {
```
Comment (CalamityAdam) [inline packages/steppp-react/src/components/Provider.tsx:39]: 🥇 100000 points for using `uselayoutEffect()`

**Review by CalamityAdam — APPROVED** [review]
gr8 work

**Review by JZDesign — APPROVED** [review]

**Pull request merged**